### DISCUSSION TITLE
latest way to remove request from imageCache?

### DISCUSSION THREAD

**Discussion by doncl** [kean/Nuke#612] [Q&A]
HI, a Nuke noob here (well, I've worked at shops that used it, but never had occasion to change any of the client code before).

I've got a legacy project that needs updating to Xcode 14 - it has a lot of (sigh) Cocoapods.  Nuke (and RxNuke) is one of them.

When updating to Xcode 14, after much, much wrangling with the Podfile, I got all the inter-dependencies to quit kvetching, and I

my Podfile.lock now says (for Nuke):    - Nuke (9.5.0)

OK, so far so good, so compiling: 

I get errors for the following line (which is 2-3 times in the codebase).
**ImagePipeline.shared.configuration.imageCache?.removeResponse(for: request)**

I've replaced it with 
**ImagePipeline.shared.configuration.imageCache?[request] = nil**

Is that correct?   I have the feeling this is incomplete, and is dangling stuff in the filesystem.   

Can anyone advise, as to the canonically correct modern way to do this?

**Comment by kean** [comment]
Please follow the [Caching](https://kean-docs.github.io/nuke/documentation/nuke/accessing-caches) guide. I no longer remember how it was done in 9.x. Please consider updating to Nuke 11.x. It's available via Swift Package Manager and Carthage (but not CocoaPods).
